### PR TITLE
Add rubygem-fog-proxmox package

### PR DESCRIPTION
**Pull request by tristanrobert**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.21
* [ ] 1.20

See Foreman's [plugin maintainer documentation](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

File: rel-eng/tito.props [hunk 1]
```diff
@@ -525,3 +526,8 @@ whitelist = katello
   rubygem-runcible
   rubygem-typhoeus
   rubygem-zest
+
+
+[rhel7]
```
Comment (ekohl) [inline rel-eng/tito.props]: This looks odd

File: packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+%{!?scl:%global pkg_name %{name}}
+
+%global gem_name fog-proxmox
+
+Name: %{?scl_prefix}rubygem-%{gem_name}
+Version: 0.7.0
+Release: 1%{?dist}
+Summary: Module for the 'Fog' gem to support Proxmox VE
+Group: Development/Languages
+License: GPL-3.0
```
Comment (ekohl) [inline packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec]:
```suggestion
License: GPLv3
```

File: packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+mkdir -p %{buildroot}%{_bindir}
+cp -pa .%{_bindir}/* \
+        %{buildroot}%{_bindir}/
+find %{buildroot}%{gem_instdir}/bin -type f | xargs chmod a+x
+
+%files
+%dir %{gem_instdir}
+%{_bindir}/console
+%{_bindir}/setup
+%{gem_instdir}/.bundle
```
Comment (ekohl) [inline packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec]: I think all of these dotfiles can be `%exclude`
```suggestion
%exclude %{gem_instdir}/.bundle
```

File: packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+%{gem_instdir}/.github
+%exclude %{gem_instdir}/.gitignore
+%{gem_instdir}/.gitlab-ci.yml
+%exclude %{gem_instdir}/.rubocop.yml
+%{gem_instdir}/.ruby-gemset
+%{gem_instdir}/.solargraph.yml
+%exclude %{gem_instdir}/.travis.yml
+%{gem_instdir}/.vscode
+%{gem_instdir}/CODE_OF_CONDUCT.md
+%{gem_instdir}/ISSUE_TEMPLATE.md
```
Comment (ekohl) [inline packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec]:
```suggestion
%exclude %{gem_instdir}/ISSUE_TEMPLATE.md
```

File: packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+%{gem_instdir}/.codeclimate.yml
+%{gem_instdir}/.github
+%exclude %{gem_instdir}/.gitignore
+%{gem_instdir}/.gitlab-ci.yml
+%exclude %{gem_instdir}/.rubocop.yml
+%{gem_instdir}/.ruby-gemset
+%{gem_instdir}/.solargraph.yml
+%exclude %{gem_instdir}/.travis.yml
+%{gem_instdir}/.vscode
+%{gem_instdir}/CODE_OF_CONDUCT.md
```
Comment (ekohl) [inline packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec]:
```suggestion
%doc %{gem_instdir}/CODE_OF_CONDUCT.md
```

File: packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        %{buildroot}%{gem_dir}/
+
+mkdir -p %{buildroot}%{_bindir}
+cp -pa .%{_bindir}/* \
+        %{buildroot}%{_bindir}/
+find %{buildroot}%{gem_instdir}/bin -type f | xargs chmod a+x
+
+%files
+%dir %{gem_instdir}
+%{_bindir}/console
```
Comment (ekohl) [inline packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec]: This sounds very generic. I'm wondering if we should `%exclude` this and `setup` below.

File: packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+%{gem_instdir}/.gitlab-ci.yml
+%exclude %{gem_instdir}/.rubocop.yml
+%{gem_instdir}/.ruby-gemset
+%{gem_instdir}/.solargraph.yml
+%exclude %{gem_instdir}/.travis.yml
+%{gem_instdir}/.vscode
+%{gem_instdir}/CODE_OF_CONDUCT.md
+%{gem_instdir}/ISSUE_TEMPLATE.md
+%license %{gem_instdir}/LICENSE
+%{gem_instdir}/SUPPORT.md
```
Comment (ekohl) [inline packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec]: Can be moved to `%files doc`
```suggestion
%doc %{gem_instdir}/SUPPORT.md
```

File: packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+%{gem_libdir}
+%exclude %{gem_cache}
+%{gem_spec}
+
+%files doc
+%doc %{gem_docdir}
+%doc %{gem_instdir}/CONTRIBUTING.md
+%doc %{gem_instdir}/CONTRIBUTORS.md
+%{gem_instdir}/Gemfile
+%{gem_instdir}/Gemfile.lock
```
Comment (ekohl) [inline packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec]:
```suggestion
%exclude %{gem_instdir}/Gemfile.lock
```

File: packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+%global gem_name fog-proxmox
+
+Name: %{?scl_prefix}rubygem-%{gem_name}
+Version: 0.7.0
+Release: 1%{?dist}
+Summary: Module for the 'Fog' gem to support Proxmox VE
+Group: Development/Languages
+License: GPL-3.0
+URL: http://github.com/fog/fog-proxmox
```
Comment (ekohl) [inline packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec]:
```suggestion
URL: https://github.com/fog/fog-proxmox
```

File: rel-eng/tito.props (hunk 1 above)
Comment (tristanrobert) [reply rel-eng/tito.props]: This is `./add_gem_package.sh rubygem-fog-proxmox scl foreman-plugins-nightly-rhel7` script I launched which did it. I think it sorts list in alphabetic order. I did not edit manually this file.

Comment (ekohl) [reply rel-eng/tito.props]: Perhaps you called it with a space and then rhel7? Normally this section isn't supposed to exist.

Comment (tristanrobert) [reply rel-eng/tito.props]: I can't remember. Ok I will start again.

Comment (ekohl) [reply rel-eng/tito.props]: Perhaps it's easier to do `git revert 2cc510d33c02e9ec42e005a2cc38e944ef378506 --no-commit && ./bump_rpm.sh packages/plugin/rubygem-fog-proxmox`

Comment (tristanrobert) [reply rel-eng/tito.props]: I have tried in a new fresh branch. I have launched `./add_gem_package.sh fog-proxmox scl foreman-plugins-nightly-rhel7` with no space in one command line. And the script did the same commits: rearrange foreman-task. Have you any idea what command I should launch instead?

Comment (tristanrobert) [reply rel-eng/tito.props]: git revert solution is not so easier: many changes to check... The issue you checked comes from `./add_gem_package.sh` script which sorts foreman-tasks. It does not break the build.

Comment (tristanrobert) [reply rel-eng/tito.props]: Sorting tito.props whitelist is here in the code: https://github.com/theforeman/foreman-packaging/blob/401a1fe288b56283be0d35a2476ad449a7f76baa/add_gem_package.sh#L71

Comment (tristanrobert) [reply rel-eng/tito.props]: I have still done the changes you asked for.

File: packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+%dir %{gem_instdir}
+%{_bindir}/console
+%{_bindir}/setup
+%exclude %{gem_instdir}/.bundle
+%exclude %{gem_instdir}/.codeclimate.yml
+%exclude %{gem_instdir}/.github
+%exclude %{gem_instdir}/.gitignore
+%exclude %{gem_instdir}/.gitlab-ci.yml
+%exclude %{gem_instdir}/.rubocop.yml
+%{gem_instdir}/.ruby-gemset
```
Comment (ekohl) [inline packages/plugins/rubygem-fog-proxmox/rubygem-fog-proxmox.spec]:
```suggestion
%exclude %{gem_instdir}/.ruby-gemset
```

**Comment by ekohl** [comment]
I've created https://github.com/theforeman/foreman-packaging/pull/3754 which maintains the changelog from previous versions and drops the binaries. They have very generic names and I doubt they actually work since they're bundler stubs. It'd be really nice if the gem itself already stripped these and other dotfiles in the build process.

**Comment by ekohl** [comment]
In 34a19b0c328fda93db1c41a0476615ab43d98564 I've re-introduced the plugin, including a cherry pick in cd7b7271348c8f3be8672ec674d220a61b7bc166 to 1.22.

**Pull request closed without merging**